src/main/java/graphql/schema/DataFetchingFieldSelectionSetImpl.java

```diff
@@ -311,18 +316,28 @@ public String getFullyQualifiedName() {
         }
 
         @Override
-        public GraphQLObjectType getObjectType() {
-            return normalizedField.getObjectType();
+        public List<GraphQLFieldDefinition> getFieldDefinitions() {
+            return normalizedField.getFieldDefinitions(schema);
+        }
+
+        @Override
+        public GraphQLOutputType getType() {
+            return normalizedField.getType(schema);
+        }
+
+        @Override
+        public List<GraphQLObjectType> getObjectTypes() {
+            return (List) this.schema.getTypes(normalizedField.getObjectTypeNames());
         }
 
         @Override
-        public GraphQLFieldDefinition getFieldDefinition() {
-            return normalizedField.getFieldDefinition();
+        public List<String> getObjectTypeNames() {
+            return new ArrayList<>(normalizedField.getObjectTypeNames());
         }
 
         @Override
         public Map<String, Object> getArguments() {
-            return normalizedField.getArguments();
+            return normalizedField.getResolvedArguments();
```

Comment:
Immutable ??

Reply:
changed to returning an in immutable Map